{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Data_Preparation.ipynb","provenance":[],"authorship_tag":"ABX9TyOS8sYwvsXrxcOgkPZrL4uZ"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","metadata":{"id":"oM61XFxm_PU-"},"source":["!pip install Bio"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"4G8cHGTm_ta_","executionInfo":{"status":"ok","timestamp":1618453532429,"user_tz":-330,"elapsed":94077,"user":{"displayName":"Sadil Chamishka","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhDLF_8ExA1AR8JO6dxWcjVTvU6IPpQzTxN9Upz=s64","userId":"18289865951350633719"}},"outputId":"8463eef4-57a7-4282-830b-924d62782e85"},"source":["import os\n","import pandas as pd\n","import numpy as np\n","import matplotlib.pyplot as plt\n","from Bio import SeqIO\n","from Bio import Entrez\n","from Bio.Align.Applications import ClustalOmegaCommandline\n","from Bio import Phylo, AlignIO\n","from Bio.Phylo.TreeConstruction import DistanceCalculator, DistanceTreeConstructor\n","from Bio.Phylo import draw\n","\n","from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Mounted at /content/drive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"fkAv-QC-FHSC"},"source":["Entrez.email='sandilchamishka@gmail.com'"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"0ki88xcgAWE4"},"source":["Accession_ids = ['GU190215.1', 'KJ473815.1', 'GQ153547.1' , 'MN996532.1' , 'KY417150.1' , 'KJ473816.1' , 'MK211376.1' , 'MG772933.1' , 'MG772934.1' , 'AY572034.1' , 'MT019529.1' , 'AY291315.1' , 'MT040334.1' , 'MT072864.1' , 'MT084071.1' , 'LC556375.1', 'MW251311' , 'MW251310.1' , 'MW251309.1' , 'MW251312']"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"62g-PPdGdnY4"},"source":["evaluated_genes = ['1ab','orf3a','orf6','orf7a','orf8','orf10','spike','envelope','membrane','nucleocapsid']\n","evaluated_genes_dic = {'1ab':[],'orf3a':[],'orf6':[],'orf7a':[],'orf8':[],'orf10':[],'spike':[],'envelope':[],'membrane':[],'nucleocapsid':[]}"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"JSbPcYfjWpBf","executionInfo":{"status":"ok","timestamp":1618456560169,"user_tz":-330,"elapsed":34582,"user":{"displayName":"Sadil Chamishka","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhDLF_8ExA1AR8JO6dxWcjVTvU6IPpQzTxN9Upz=s64","userId":"18289865951350633719"}},"outputId":"18cbd988-b57c-44a9-fd24-c1294e2bba2e"},"source":["lst_record = []\n","for id in Accession_ids:\n","  handle = Entrez.efetch(db='nucleotide', id=id, rettype='gb')\n","  record = SeqIO.read(handle, \"genbank\")\n","  handle.close()\n","  print(id)\n","  print(record.annotations['source'])\n","  print(\"***************\")\n","  lst_record.append(record)\n","\n","SeqIO.write(lst_record, \"/content/drive/My Drive/BioinformaticsProject/Experiment1/cov_sample.fasta\", \"fasta\")"],"execution_count":null,"outputs":[{"output_type":"stream","text":["GU190215.1\n","Bat coronavirus BM48-31/BGR/2008\n","***************\n","KJ473815.1\n","BtRs-BetaCoV/GX2013\n","***************\n","GQ153547.1\n","Bat SARS coronavirus HKU3-12\n","***************\n","MN996532.1\n","Bat coronavirus RaTG13\n","***************\n","KY417150.1\n","Bat SARS-like coronavirus\n","***************\n","KJ473816.1\n","BtRs-BetaCoV/YN2013\n","***************\n","MK211376.1\n","Coronavirus BtRs-BetaCoV/YN2018B\n","***************\n","MG772933.1\n","Bat SARS-like coronavirus\n","***************\n","MG772934.1\n","Bat SARS-like coronavirus\n","***************\n","AY572034.1\n","Civet SARS CoV 007/2004\n","***************\n","MT019529.1\n","Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n","***************\n","AY291315.1\n","SARS coronavirus Frankfurt 1\n","***************\n","MT040334.1\n","Pangolin coronavirus\n","***************\n","MT072864.1\n","Pangolin coronavirus\n","***************\n","MT084071.1\n","Pangolin coronavirus\n","***************\n","LC556375.1\n","Severe acute respiratory syndrome-related coronavirus\n","***************\n","MW251311\n","Bat coronavirus RacCS264\n","***************\n","MW251310.1\n","Bat coronavirus RacCS253\n","***************\n","MW251309.1\n","Bat coronavirus RacCS224\n","***************\n","MW251312\n","Bat coronavirus RacCS271\n","***************\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"xNdmu64EeASm"},"source":["for id in Accession_ids:\n","  handle = Entrez.efetch(db='nucleotide', id=id, rettype='gb')\n","  record = SeqIO.read(handle, \"genbank\")\n","  handle.close()\n","  for feature in record.features:\n","    if feature.type == 'CDS':\n","      gene = feature.qualifiers['product'][0].lower()\n","      for ev_gene in evaluated_genes:\n","        if ev_gene in gene or \"surface glycoprotein\" in gene:\n","          entry = f\">{record.annotations['source']} ({id}) \\n\"\n","          entry = entry.replace(\" \",\"-\")\n","          entry += record.seq[feature.location.start : feature.location.end]\n","          if \"surface glycoprotein\" in gene:\n","            evaluated_genes_dic[\"spike\"].append(entry)\n","          evaluated_genes_dic[ev_gene].append(entry)\n","          break"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"40q2eoaBE9Jf"},"source":["for key, value in evaluated_genes_dic.items():\n","  data = \"\"\n","  for seq in evaluated_genes_dic[key]:\n","    data += seq + \"\\n\"\n","  file = open(f'/content/drive/My Drive/BioinformaticsProject/Experiment1/Homologous_Sequences/{key}.fasta', \"w\")\n","  file.write(str(data))\n","  file.close()"],"execution_count":null,"outputs":[]}]}